packages/firestore/src/api/user_data_reader.ts

```diff
@@ -398,6 +405,148 @@ export function parseSetData(
   );
 }
 
+export class DeleteFieldValueImpl extends FieldValue {
```

Comment:
A possible thought to follow up on layering: would it make sense to explicitly break out common code and the classic API as separate paths? That way you could impose layering checks, e.g.

  * lite -> common
  * exp -> (lite, common)
  * classic -> (exp, common)

or similar?

Reply:
Once all the dust has settled, I would like to move the /exp and /lite files under /src. They only contain API code and the nested imports are not pretty. This is something I will consider then.